openeo/metadata.py

```diff
@@ -540,6 +540,21 @@ def metadata_from_stac(url: str) -> CubeMetadata:
 
     # TODO move these nested functions and other logic to _StacMetadataParser
 
+    def get_temporal_metadata(spec: dict, complain: Callable[[str], None] = warnings.warn) -> TemporalDimension:
```

Comment:
as noted, I would not define this helper to work on dictionaries, but with pystac objects directly, to leverage all its existing  validation and discovery functionality